app/uk/gov/hmrc/nisp/models/NIRecordTaxYear.scala

```diff
@@ -16,14 +16,33 @@
 
 package uk.gov.hmrc.nisp.models
 
+import org.joda.time.LocalDate
 import play.api.libs.json.Json
 import uk.gov.hmrc.nisp.utils.Constants
 
 case class NIRecordTaxYear(taxYear: Int, qualifying: Boolean,classOneContributions: BigDecimal,
                            classTwoCredits: Int, classThreeCredits: Int, otherCredits: Int,
                            classThreePayable: Option[BigDecimal], classThreePayableBy: Option[NpsDate],
                            classThreePayableByPenalty: Option[NpsDate], payable: Boolean, underInvestigation: Boolean) {
+
   val displayableTaxYear: String = s"$taxYear-${(taxYear + 1).toString.substring(Constants.shortYearStartCharacter,Constants.shortYearEndCharacter)}"
+
+}
+
+
+trait isCutOffDate {
+
+  def cutOffDate(b: Option[NpsDate], c: LocalDate): Boolean = { 
```

Comment:
This is a small helper function, so we should not care about logical names. As long as it does the job. However I will change it as per your request.